src/kbmod/trajectory_generator.py

```diff
@@ -314,25 +315,153 @@ def __init__(self, v_arr, ang_arr, average_angle, *args, **kwargs):
+                )
+                average_angle = work_unit.compute_ecliptic_angle()
 
         ang_min = average_angle - ang_arr[0]
         ang_max = average_angle + ang_arr[1]
-        super().__init__(v_arr[2], v_arr[0], v_arr[1], ang_arr[2], ang_min, ang_max, *args, **kwargs)
+        super().__init__(v_arr[2], v_arr[0], v_arr[1], ang_arr[2], ang_min, ang_max, **kwargs)
+
+
+class EclipticCenteredSearch(TrajectoryGenerator):
+    """Search a grid defined by velocities and angles relative to the ecliptic.
+
+    Attributes
+    ----------
+    ecliptic_angle : `float`
+        The angle to use for the ecliptic in the image (in the units defined in ``angle_units``).
+    velocities : `list`
+        A triplet of the minimum velocity to use (in pixels per day), and the maximum velocity
+        magnitude (in pixels per day), and the number of velocity steps to try.
+    angles : `list`
+        A triplet of the minimum angle offset (in radians), and the maximum angle offset (in radians),
+        and the number of angles to try.
+    min_ang : `float`
+        The minimum search angle in the image (ecliptic_angle + min_ang_offset) in radians.
+    max_ang : `float`
+        The maximum search angle in the image (ecliptic_angle + max_ang_offset) in radians.
+    """
+
+    def __init__(
+        self,
+        velocities=[0.0, 0.0, 0],
+        angles=[0.0, 0.0, 0],
+        angle_units="radians",
+        given_ecliptic=None,
+        work_unit=None,
+        **kwargs,
+    ):
+        """Create a class EclipticCenteredSearch.
+
+        Parameters
+        ----------
+        velocities : `list`
+            A triplet of the minimum velocity to use (in pixels per day), and the maximum velocity
+            magnitude (in pixels per day), and the number of velocity steps to try.
+        angles : `list`
+            A triplet of the minimum angle offset (in the units defined in ``angle_units``), the maximum
+            angle offset (in the units defined in ``angle_units``), and the number of angles to try.
+        angle_units : `str`
+            The units for the angle. Must be one of radians or degrees.
+            Default: 'radians'
+        given_ecliptic : `float`, optional
+            An override for the ecliptic as given in the config (in the units defined in
+            ``angle_units``). This angle takes precedence over ``computed_ecliptic``.
+        work_unit : `WorkUnit`, optional
+            A WorkUnit to provide additional information about the data that
+            can be used to derive parameters that depend on the input.
+        """
+        super().__init__(**kwargs)
+
+        if given_ecliptic is not None:
+            if angle_units[:3] == "deg":
+                ecliptic_angle = given_ecliptic * (math.pi / 180.0)
+            elif angle_units[:3] == "rad":
+                ecliptic_angle = given_ecliptic
+            else:
```

Comment:
I'm not sure I fully understand why we can't check for quantity and always cast to radians and pixel/day internally and then strip the unit? I'm not going to be a stickler about it, this is fine, but I am curious.